pkg/cloud/services/networking/network.go

```diff
@@ -369,3 +370,27 @@ func GetSubnetsByFilter(networkClient *gophercloud.ServiceClient, opts subnets.L
 	}
 	return snets, nil
 }
+
+func (s *Service) DeleteOrphanedPorts() error {
+	portList, err := ports.List(s.client, ports.ListOpts{ProjectID: s.projectID}).AllPages()
+	if err != nil {
+		return fmt.Errorf("error listing ports: %v", err)
+	}
+
+	orphanedPorts, err := ports.ExtractPorts(portList)
+	if err != nil {
+		return fmt.Errorf("error extracting ports: %v", err)
+	}
+
+	for _, orphanedPort := range orphanedPorts {
+		if orphanedPort.DeviceOwner != "" {
+			continue
+		}
+
```

Comment:
I'd think we at least log this behavior

Reply:
Yup I"ll add log statements where we do (i.e. delete) something